pkg/app/appdisc/factory.go

```diff
@@ -0,0 +1,77 @@
+package appdisc
+
+import (
+	"strings"
+	"time"
+
+	"github.com/SkycoinProject/dmsg/cipher"
+	"github.com/SkycoinProject/skycoin/src/util/logging"
+	"github.com/sirupsen/logrus"
+
+	"github.com/SkycoinProject/skywire-mainnet/pkg/app/appcommon"
+	"github.com/SkycoinProject/skywire-mainnet/pkg/proxydisc"
+	"github.com/SkycoinProject/skywire-mainnet/pkg/skyenv"
+)
+
+// Factory creates appdisc.Updater instances based on the app name.
+type Factory struct {
+	Log            logrus.FieldLogger
```

Comment:
Shouldn't these fields be unexported for safety?

Reply:
My thoughts are that `appdisc.Factory` acts as a *config* or sorts that generates discovery updaters. These fields are only read for generating such discovery updaters.

Hope that is okay?